source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1758 @@
+  is-sup x = pr₂ (h x)
+
+  is-upper-bound-↓ : (x : ⟨ L ⟩)
+                   → (x is-an-upper-bound-of (↓ᴮ x , ↓ᴮ-inclusion x)) holds
+  is-upper-bound-↓ x = pr₁ (is-sup x)
+
+  is-least-upper-bound-↓ : (x : ⟨ L ⟩)
+                         → ((u' , _) : upper-bound (↓ᴮ x , ↓ᴮ-inclusion x))
+                         → (x ≤ u') holds
+  is-least-upper-bound-↓ x = pr₂ (is-sup x)
+
+  _≤ᴮ_ : (b : B) (x : ⟨ L ⟩) → 𝓥  ̇
+  b ≤ᴮ x = pr₁ (≤-is-small x b)
+
+  _≤ᴮ_-≃-_≤_ : {b : B} {x : ⟨ L ⟩} → (b ≤ᴮ x) ≃ ((q b) ≤ x) holds
+  _≤ᴮ_-≃-_≤_ {b} {x} = pr₂ (≤-is-small x b)
+
+  _≤ᴮ_-to-_≤_ : {b : B} {x : ⟨ L ⟩} → (b ≤ᴮ x) → ((q b) ≤ x) holds
+  _≤ᴮ_-to-_≤_ = ⌜ _≤ᴮ_-≃-_≤_ ⌝
+
+  _≤_-to-_≤ᴮ_ : {b : B} {x : ⟨ L ⟩} → ((q b) ≤ x) holds → (b ≤ᴮ x)
+  _≤_-to-_≤ᴮ_ = ⌜ _≤ᴮ_-≃-_≤_ ⌝⁻¹
+
+  _≤ᴮ_-is-prop-valued : {b : B} {x : ⟨ L ⟩} → is-prop (b ≤ᴮ x)
+  _≤ᴮ_-is-prop-valued {b} {x} =
+   equiv-to-prop _≤ᴮ_-≃-_≤_ (holds-is-prop ((q b) ≤ x))
+
+  small-↓ᴮ : ⟨ L ⟩ → 𝓥  ̇
+  small-↓ᴮ x = Σ b ꞉ B , b ≤ᴮ x
+
+  small-↓ᴮ-inclusion : (x : ⟨ L ⟩) → small-↓ᴮ x → ⟨ L ⟩
+  small-↓ᴮ-inclusion x = q ∘ pr₁
+
+  small-↓ᴮ-≃-↓ᴮ : {x : ⟨ L ⟩} → small-↓ᴮ x ≃ ↓ᴮ x
+  small-↓ᴮ-≃-↓ᴮ {x} = Σ-cong' P Q e
+   where
+    P : B → 𝓥  ̇
+    P b = b ≤ᴮ x
+    Q : B → 𝓦  ̇
+    Q b = ((q b) ≤ x) holds
+    e : (b : B) →  b ≤ᴮ x ≃ ((q b) ≤ x) holds
+    e b = _≤ᴮ_-≃-_≤_ {b} {x}
+
+  ↓ᴮ-is-small : {x : ⟨ L ⟩} → ↓ᴮ x is 𝓥 small
+  ↓ᴮ-is-small {x} = (small-↓ᴮ x , small-↓ᴮ-≃-↓ᴮ {x})
+
+  is-sup'ᴮ : (x : ⟨ L ⟩) → x ＝ ⋁ (small-↓ᴮ x , small-↓ᴮ-inclusion x)
+  is-sup'ᴮ x = ≃-families-＝-sup
+               x
+               (⋁ (small-↓ᴮ x , small-↓ᴮ-inclusion x))
+               (is-sup x)
+               (is-lub-for L ((small-↓ᴮ x , small-↓ᴮ-inclusion x)))
+   where
+    open Equivalent-Families-have-same-Join L (↓ᴮ x)
+                                              (small-↓ᴮ x)
+                                              small-↓ᴮ-≃-↓ᴮ
+                                              (↓ᴮ-inclusion x)
+                                              hiding (⋁_)
+
+  is-supᴮ : (x : ⟨ L ⟩)
+          → (x is-lub-of (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds
+  is-supᴮ x =
+    transport (λ z → (z is-lub-of (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds)
+              (is-sup'ᴮ x ⁻¹)
+              (is-lub-for L ((small-↓ᴮ x , small-↓ᴮ-inclusion x)))
+
+  is-upper-boundᴮ : (x : ⟨ L ⟩)
+                  → (x is-an-upper-bound-of
+                       (small-↓ᴮ x , small-↓ᴮ-inclusion x)) holds
+  is-upper-boundᴮ x = pr₁ (is-supᴮ x)
+
+  is-least-upper-boundᴮ : (x : ⟨ L ⟩)
+                        → ((u' , _) : upper-bound
+                                      (small-↓ᴮ x , small-↓ᴮ-inclusion x))
+                        → (x ≤ u') holds
+  is-least-upper-boundᴮ x = pr₂ (is-supᴮ x)
+
+\end{code}
+
+Now we construct the least closed subset of an inductive definition as a QIT.
+Since HIT's are not native in Agda we postulate the existence of such a type
```

Comment:
It is confusing to say that you "postulate" something, because this would imply that you are using the Agda keyword `postulate`, which are not (in fact, because we are under `--safe`, you wouldn't be able to do that). This could be fixed by saying "we assume the existence of such a type", but even this is not an optimal phrasing. Also saying "axiomatic" is a little bit confusing, because you are not using axioms. Have a go rephrasing the whole paragraph, and if you are stuck I can help you.

Reply:
I gave it a shot.